src/components/date/date.component.tsx

```diff
@@ -85,7 +143,7 @@ const DateInput = React.forwardRef(
       dateFnsLocale,
     ]);
     const { inputRefMap, setInputRefMap } = useContext(DateRangeContext);
-    const inputName = dataElement?.split("-")[0];
+    const inputName = dataElement?.split("-")[0] || "start";
```

Comment:
I don't think this will create any issues tbh, but I don't really like the approach so I'm going to make a small refactor here and ensure that the it uses a unique id instead rather that the dataElement

Reply:
I'll make this change as a new commit which will go after the ones refactoring the files to TS